### PR TITLE
Better SDL2 Audio Init Error Logging

### PR DESCRIPTION
**Pull request by IsaacMarovitz**
Better logging for SDL2 audio init errors to better diagnose the greyed out option issue

### CODE REVIEW

File: Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs [hunk 1]
```diff
@@ -112,13 +113,17 @@ internal static uint OpenStream(SampleFormat requestedSampleFormat, uint request
 
             if (device == 0)
             {
+                string errorMessage = $"SDL2 open audio device initialization failed with error \"{SDL_GetError()}\"";
```
Comment (marysaka) [inline Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs:116]:
```suggestion
                string errorMessage = $"SDL2 open audio device initialization failed with error \"{SDL_GetError()}\"";

```

File: Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs [hunk 2]
```diff
@@ -112,13 +113,17 @@ internal static uint OpenStream(SampleFormat requestedSampleFormat, uint request
...
+                Logger.Error?.Print(LogClass.Application, errorMessage);
                 return 0;
             }
 
             bool isValid = got.format == desired.format && got.freq == desired.freq && got.channels == desired.channels;
 
             if (!isValid)
             {
+                string errorMessage = $"SDL2 open audio device is not valid";
```
Comment (marysaka) [inline Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs:125]:
```suggestion
                string errorMessage = $"SDL2 open audio device is not valid";

```

**Review by marysaka — CHANGES_REQUESTED** [review]

File: Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs [hunk 3]
```diff
@@ -112,13 +113,19 @@ internal static uint OpenStream(SampleFormat requestedSampleFormat, uint request
...
+
+                Logger.Error?.Print(LogClass.Application, errorMessage);
                 return 0;
             }
 
             bool isValid = got.format == desired.format && got.freq == desired.freq && got.channels == desired.channels;
 
             if (!isValid)
             {
+                string errorMessage = $"SDL2 open audio device is not valid";
```
Comment (marysaka) [inline Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs:126]:
```suggestion
                string errorMessage = $"SDL2 audio device is invalid!";
```

**Review by marysaka — CHANGES_REQUESTED** [review]

File: Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs [hunk 4]
```diff
@@ -112,13 +113,19 @@ internal static uint OpenStream(SampleFormat requestedSampleFormat, uint request
...
+
+                Logger.Error?.Print(LogClass.Application, errorMessage);
```
Comment (AcK77) [inline Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs:128]: `errorMessage` can be removed (and the string interpolation too).

```suggestion
                Logger.Error?.Print(LogClass.Application, "SDL2 open audio device is not valid");
```

File: Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs [hunk 5]
```diff
@@ -112,13 +113,19 @@ internal static uint OpenStream(SampleFormat requestedSampleFormat, uint request
 
             if (device == 0)
             {
+                string errorMessage = $"SDL2 open audio device initialization failed with error \"{SDL_GetError()}\"";
+
+                Logger.Error?.Print(LogClass.Application, errorMessage);
```
Comment (AcK77) [inline Ryujinx.Audio.Backends.SDL2/SDL2HardwareDeviceDriver.cs:118]:
```suggestion
                Logger.Error?.Print(LogClass.Application, 
                    $"SDL2 open audio device initialization failed with error \"{SDL_GetError()}\"");
```
Add a new extra empty line before the `return` too

**Review by AcK77 — CHANGES_REQUESTED** [review]

**Review by AcK77 — APPROVED** [review]

**Pull request merged**